scripts/entity/components/attack_components/base_attack_component.gd

```diff
@@ -96,8 +93,21 @@ func _add_target(_body: Node2D):
 		_update_targets = true
 		update_closest_target()
 
+func _get_current_target_or_direction():
+	if is_instance_valid(_current_target):
+		return _current_target
+	else:
+		return Vector2.RIGHT
```

Comment:
Would be nice to make this whichever direction we're facing rather than just right.

Reply:
Targeting and shooting direction fixed. animations can be fixed with the moth